featureflags.go

```diff
@@ -327,12 +328,31 @@ func (poller *FeatureFlagsPoller) run() {
 func (poller *FeatureFlagsPoller) fetchNewFeatureFlags() {
 	personalApiKey := poller.personalApiKey
 	headers := http.Header{"Authorization": []string{"Bearer " + personalApiKey}}
-	res, cancel, err := poller.localEvaluationFlags(headers)
+
+	// Read current ETag under lock
+	poller.mutex.RLock()
+	currentEtag := poller.flagsEtag
+	poller.mutex.RUnlock()
+
+	res, cancel, err := poller.localEvaluationFlags(headers, currentEtag)
 	if err != nil {
 		poller.Logger.Errorf("Unable to fetch feature flags: %s", err)
 		return
 	}
 	defer cancel()
+	defer res.Body.Close()
```

Comment:
Moved this call here because before:
- If status was `402` (quota limit) → returned early, body never closed (leak!) 
- If status was not `200` → returned early, body never closed (leak!)